### PR TITLE
nx-libs, x2goclient and x2goserver

### PR DESCRIPTION
**Pull request by eoli3n**
Closes https://github.com/void-linux/void-packages/issues/9779 and https://github.com/void-linux/void-packages/issues/2091

From https://github.com/renatoaguiar/void-packages/tree/x2goclient
Ping @renatoaguiar, I sent you a mail, if you want to be the maintainer, just ask.

# nx-libs
- [x] Changed license
- [x] Updated maintainer
- [x] Changed nx-libs source to https://github.com/ArcticaProject/nx-libs/
- [x] Updated to latest version
- [x] Set ``python_version``
- [x] Added libs to ``common/shlib``
- [x] Patch for musl, and bigfont
- [x] Disable cross compilation for the current version
- [x] Discussion to fix some limitations upstream in next versions : https://github.com/ArcticaProject/nx-libs/issues/975
  - Drop imake for automake: crossbuild for arm -> WIP https://github.com/ArcticaProject/nx-libs/pull/946
  - Patch mesa and bigfont: native musl builds -> WIP https://github.com/ArcticaProject/nx-libs/pull/976

The current version is ok to be merged.

# x2goclient
- [x] Changed license
- [x] Updated maintainer
- [x] Updated to latest version
- [ ] Disable cross compilation until nx-libs update ?
# x2goserver

- [x] Create template
- [x] Patch ``/usr/sbin`` references
- [ ] Init script for ``$PREFIX/bin/x2gocleansessions``
- [ ] Write runit service
- [ ] Disable cross compilation until nx-libs update ?

# Note
- https://git.alpinelinux.org/aports/tree/community/nx-libs/APKBUILD
- https://git.alpinelinux.org/aports/tree/community/x2goserver/APKBUILD

### CODE REVIEW

**Comment by yopito** [comment]
hello, I've retrieved an old WIP branch of mine: https://github.com/yopito/void-packages/tree/WIP.x2go
Might help you to package it.
Feel free to reuse its content.

**Comment by ahesford** [comment]
Cf. https://github.com/ahesford/void-packages/commit/2653816f13e501d10657b9c1094c8ecbe4004c9b

File: srcpkgs/x2goclient/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'x2goclient'
+pkgname=x2goclient
+version=4.1.2.2
+revision=1
+build_style=gnu-makefile
+make_build_target="build_client build_man"
+make_install_target="install_client install_man"
+hostmakedepends="pkg-config qt-devel-tools"
+makedepends="qt-devel libldap-devel libssh-devel libXpm-devel cups-devel"
```
Comment (Johnnynator) [inline srcpkgs/x2goclient/template:9]: Do use qt5, it should also be supported (at least nix is using it). Qt4 will be dropped some time soon (and is ages EOL)

**Comment by Johnnynator** [comment]
It also needs `libtirpc-devel`
nx-libs fails to compile with
```
utils.c: In function 'LockServer':
utils.c:382:35: warning: 'sprintf' may write a terminating nul past the end of the destination [-Wformat-overflow=]
  382 |   (void) sprintf(pid_str, "%10ld\n", (long)getpid());
      |                                   ^
utils.c:382:10: note: 'sprintf' output between 12 and 13 bytes into a destination of size 12
  382 |   (void) sprintf(pid_str, "%10ld\n", (long)getpid());
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC strlcat.c
rpcauth.c:47:10: fatal error: rpc/rpc.h: No such file or directory
   47 | #include <rpc/rpc.h>
      |          ^~~~~~~~~~~
compilation terminated.
failed command: gcc -c -g -O3 -fno-strict-aliasing -std=c99 -Wall -Wpedantic -Wpointer-arith -Wshadow -Wundef -I. -I../include -I../../../exports/include/nx-X11 -I../../../include/extensions -I../../../programs/Xserver/Xext -I../../../programs/Xserver/render -I../../../lib/Xau -I../../../../nxcomp -I/usr/include/pixman-1 -I../../.. -I../../../exports/include -Dlinux -D__amd64__ -D_POSIX_C_SOURCE=199309L -D_POSIX_SOURCE -D_XOPEN_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -DSHAPE -DXINPUT -DXKB -DXCSECURITY -DXF86BIGFONT -DDPMSExtension -DRENDER -DRANDR -DPANORAMIX -DXFIXES -DDAMAGE -DCOMPOSITE -DGCCUSESGAS -DAVOID_GLYPHBLT -DPIXPRIV -DSINGLEDEPTH -DXvExtension -DXFree86Server -DXvMCExtension -DBUILDDEBUG -DXResExtension -DX_BYTE_ORDER=X_LITTLE_ENDIAN -DXORG_VERSION_CURRENT=(((6) * 10000000) + ((9) * 100000) + ((0) * 1000) + 0) -D_XSERVER64 -DNDEBUG -DHAVE_SETITIMER=1 -DHAS_XFONT2 -DFUNCPROTO=15 -DNARROWPROTO -DUNIXCONN -DTCPCONN -DHAS_STICKY_DIR_BIT -DHAS_FCHOWN -DIPv6 -DHASXDMAUTH -DSECURE_RPC -DDDXOSINIT -DSERVER_LOCK -DDDXOSFATALERROR -DDDXOSVERRORF -DCLIENTIDS -DBUILDERADDR="xorg@lists.freedesktop.org" -DFAIL_HARD -DXTRANS_SEND_FDS=0 -DNX_TRANS_SOCKET -DNX_TRANS_AUTH -DNX_TRANS_FOPEN -DNX_TRANS_SLEEP -DNX_TRANS_EXIT -DNX_TRANS_WAKEUP -DNXAGENT_SERVER rpcauth.c
make[7]: *** [Makefile:531: rpcauth.o] Error 1
make[7]: *** Waiting for unfinished jobs....
```

File: srcpkgs/x2goclient/template (hunk 1 above)
Comment (eoli3n) [reply srcpkgs/x2goclient/template]: Done

**Comment by eoli3n** [comment]
> It also needs `libtirpc-devel`
> nx-libs fails to compile with

@Johnnynator  As you said on IRC, you found this : https://github.com/ArcticaProject/nx-libs/blob/3.6.x/nx-libs.spec#L437
Then, I don't know how to force this with the actual template.

**Comment by eoli3n** [comment]
I need some help to finish this, ping @ahesford @Johnnynator @

**Comment by eoli3n** [comment]
Thanks @Johnnynator

**Comment by eoli3n** [comment]
What about ``x2goserver`` ? No need to create a runit service in the package ?

**Comment by Anachron** [comment]
Wohoo! Thanks for packaging this. 
I've closed my issue manually because it seems not to have been auto closed on merge.

**Comment by cultab** [comment]
running x2godbadmin returns this.
```
Can't locate Try/Tiny.pm in @INC (you may need to install the Try::Tiny module) (@INC contains: /usr/lib/perl5/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/core_perl /usr/share/perl5/core_perl) at /usr/local/sbin/x2godbadmin line 29.
BEGIN failed--compilation aborted at /usr/local/sbin/x2godbadmin line 29.
```

I think the` perl-Try-Tiny` package needs to be added in the dependencies, correct me if I'm wrong or if this is not the place to share this.

**Comment by ericonr** [comment]
@cultab please open a separate issue about this or a PR to fix it :)

**Pull request merged**